pkg/controller/cloudantv1/cloudantdatabase.go

```diff
@@ -0,0 +1,213 @@
+	// is there anything returned in databaseinformation that could be used for state or 
+	// should I define state similar to the way I did for topic
+	// I don't know what the options for updateseq are right now so I'm leaving "active",
+	// "inactive", and "provisioning", ??
+	if !(reference.FromPtrValue(instance.UpdateSeq) == "active" ||
+		reference.FromPtrValue(instance.UpdateSeq) == "inactive" ||
+		reference.FromPtrValue(instance.UpdateSeq) == "provisioning") {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+
+	currentSpec := cr.Spec.ForProvider.DeepCopy()
+	if err = ibmccdb.LateInitializeSpec(&cr.Spec.ForProvider, instance); err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+	}
+	if !cmp.Equal(currentSpec, &cr.Spec.ForProvider) {
+		// if there is no update what should be called here ??
+		if err := c.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrManagedUpdateFailed)
+		}
+	}
+
+	cr.Status.AtProvider, err = ibmccdb.GenerateObservation(instance)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrGenObservation)
+	}
+
+	switch cr.Status.AtProvider.UpdateSeq {
+	case "active":
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+	case "inactive":
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	case "provisioning":
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Unavailable())
+	}
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	upToDate, err := ibmccdb.IsUpToDate(&cr.Spec.ForProvider, instance, c.logger)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, ibmc.ErrCheckUpToDate)
+	}
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: nil,
+	}, nil
+}
+
+func (c *cloudantdatabaseExternal) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mg.(*v1alpha1.CloudantDatabase)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotCloudantDatabase)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Creating())
+	resInstanceOptions := &cv1.PutDatabaseOptions{}
+	if err := ibmccdb.GenerateCreateCloudantDatabaseOptions(cr.Spec.ForProvider, resInstanceOptions); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateCloudantDatabaseOpts)
+	}
+
+	_, _, err := c.client.CloudantV1().PutDatabase(resInstanceOptions)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreateCloudantDatabase)
+	}
+
+	meta.SetExternalName(cr, cr.Spec.ForProvider.Db)
+
+	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+}
+
+func (c *cloudantdatabaseExternal) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	// what should I do here if there is no update ??
```

Comment:
I think we should ensure that the `upToDate, err := ibmccdb.IsUpToDate` in Observe() always return true, then this Update() will never be invoked, I think it's fine then to have this method empty as you have.